templates/app/helpers/spree/solidus_starter_frontend_helper.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Spree
+  module SolidusStarterFrontendHelper
+    def generate_address_id(form, input_name)
+      "#{form.object_name.gsub(/[\[\]]/, '[' => '_', ']' => '_')}#{input_name}"
+    end
```

Comment:
Temporarily extracted `generate_address_id` helper. Will see in a future PR if this helper can be removed by using `ActionView::Helpers::FormBuilder` methods in the form.